www/source/blog/2017-08-09-Habitat-plans-for-net-full-iis-apps.md

```diff
@@ -0,0 +1,301 @@
+
+```powershell
+try {
+    Write-Host "{{pkg.name}} is running"
+    $running = $true
+    while($running) {
+        Start-Sleep -Seconds 1
+        $resp = Invoke-WebRequest "http://localhost:{{cfg.port}}/{{cfg.app_name}}" -Method Head
+        if($resp.StatusCode -ne 200) { $running = $false }
+    }
+}
+catch {
+    Write-Host "{{pkg.name}} HEAD check failed"
+}
+finally {
+    # Add any cleanup here which will run after supervisor stops the service
+    Write-Host "{{pkg.name}} is stoping..."
+    ."$env:SystemRoot\System32\inetsrv\appcmd.exe" stop apppool "{{cfg.app_pool}}"
+    ."$env:SystemRoot\System32\inetsrv\appcmd.exe" stop site "{{cfg.site_name}}"
+    Write-Host "{{pkg.name}} has stopped"
+}
+```
+
+As you can see, we simply call `Invoke-Webrequest` to send a `HEAD` request to our app. If the response is not a `200`, then we assume things have gone horribly wrong and exit the loop. Further, if the supervisor stops our service (`hab svc stop mwrock/hab-sln`) execution will move to the `finally` block. Here we tell IIS to shut down our application.
+
+So lets run our application. Assuming we already built it in a studio by entering a Wiondows studio via `hab studio enter -w` from the root of our solution and then running `build ./hab-sln/` to build the hab-sln project. We can stream the supervisor log and start our service:
+
+```
+[HAB-STUDIO] Habitat:\src> Get-SupervisorLog
+[HAB-STUDIO] Habitat:\src> hab svc start mwrock/hab-sln
+hab-sup(MN): Supervisor starting mwrock/hab-sln. See the Supervisor output for more details.
+```
+
+Our supervisor log should show:
+
+```
+hab-sup(MR): Supervisor Member-ID 883fe61fe11b4e64b20245adc9830bed
+hab-sup(MR): Starting gossip-listener on 0.0.0.0:9638
+hab-sup(MR): Starting http-gateway on 0.0.0.0:9631
+hab-sup(MR): Starting mwrock/hab-sln
+hab-sln.default(SR): Hooks recompiled
+default(CF): Updated website.ps1 945a1ab52bb3ad41211cd77625dd870244be1a81b822e7db171795c4f22ade86
+hab-sln.default(SR): Configuration recompiled
+hab-sln.default(SR): Initializing
+hab-sln.default(SV): Starting service as user=matt, group=
+hab-sln.default(O): Ensuring xWebAdministration DSC resources are installed...
+hab-sln.default(O): Compiling DSC mof...
+hab-sln.default(O): WARNING: The names of some imported commands from the module
+hab-sln.default(O): 'PSDesiredStateConfiguration' include unapproved verbs that might make them less
+hab-sln.default(O): discoverable. To find the commands with unapproved verbs, run the Import-Module
+hab-sln.default(O): command again with the Verbose parameter. For a list of approved verbs, type
+hab-sln.default(O): Get-Verb.
+hab-sln.default(O): Applying DSC configuration...
+hab-sln.default(O):
+hab-sln.default(O): hab-sln is running
+```
+
+and browsing to `http://localhost:8099/hab_app` should bring up our ASP.NET web site. Now lets go crazy and change the web site port:
```

Comment:
"lets" => "let's"

Also, the sentence starting with "Now" seems like it should be the start of a new paragraph.